pkgs/tools/games/gog-downloader/default.nix

```diff
@@ -0,0 +1,40 @@
+{ stdenv, fetchurl, makeWrapper, lib, php }:
+
+let
+  pname = "gog-downloader";
+  version = "1.2.3";
+  hash = "0kv09vmrz6dcjvaf2sw89cpb14wdr01r1kkzg84szfxwzr5bcc9p";
+in
+stdenv.mkDerivation {
+  inherit pname version hash;
```

Comment:
Why? It was intentionally above to have all the often changeable configuration above everything else and visible at the first glance.

Reply:
I think we should use the `finalAttrs` pattern here, which is made to override everything properly.

Why is this pattern preferred to `rec` ?

Let's take this simple code example:

```nix
mkDerivation rec {
   foo = 1;
   bar = foo + 1;
}
```

and then `.overrideAttrs(old: { foo = 2; })`, you'll get `{ foo = 2; bar = 2; }` while with `finalAttrs` pattern, it would work correctly because it's a real fixed point.

Let me share a couple of useful links regarding the `finalAttrs` pattern:

1. History: https://discourse.nixos.org/t/avoid-rec-expresions-in-nixpkgs/8293
2. Documentation: https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes
3. Recent example of implementation: https://github.com/NixOS/nixpkgs/compare/17f96f7b978e61576cfe16136eb418f74fab9952..9e6ea843e473d34d4f379b8b0d8ef0425a06defe

Feel free to reach out if you need some assistance.